src_c/rect_impl.h

```diff
@@ -894,6 +900,41 @@ RectExport_moveIp(RectObject *self, PyObject *const *args, Py_ssize_t nargs)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+RectExport_moveTo(RectObject *self, PyObject *const *args, Py_ssize_t nargs,
+                  PyObject *kwnames)
+{
+    PyObject *rect = RectExport_copy(self, NULL);
+
+    if (nargs > 0) {
+        Py_DECREF(rect);
+        return RAISE(PyExc_TypeError, "move_to takes no positional arguments");
+    }
+
+    if (rect) {
+        if (kwnames) {
+            Py_ssize_t i, sequence_len;
+            PyObject **sequence_items;
+            sequence_items = PySequence_Fast_ITEMS(kwnames);
+            sequence_len = PySequence_Fast_GET_SIZE(kwnames);
+
+            for (i = 0; i < sequence_len; ++i) {
+                if ((PyObject_SetAttr(rect, sequence_items[i], args[i]) ==
+                     -1)) {
+                    Py_DECREF(rect);
+                    return NULL;
+                }
+            }
+        }
+        else if (nargs == 0) {
+            Py_DECREF(rect);
+            return RAISE(PyExc_TypeError,
+                         "move_to cannot be called without arguments");
+        }
+    }
+    return rect;
```

Comment:
You can remove a lot of indentation levels by inverting the conditions. It would make code much more readable. You'd have to add another condition for when non keyword args are passed tho.
```suggestion
    if (!rect) {
        return NULL;
    }
    if (!kwnames) {
        Py_DECREF(rect);
        return RAISE(PyExc_TypeError,
                     "move_to cannot be called without a keyword argument");
    }
    
    Py_ssize_t i, sequence_len;
    PyObject **sequence_items;
    sequence_items = PySequence_Fast_ITEMS(kwnames);
    sequence_len = PySequence_Fast_GET_SIZE(kwnames);

    for (i = 0; i < sequence_len; ++i) {
        if ((PyObject_SetAttr(rect, sequence_items[i], args[i]) ==
             -1)) {
            Py_DECREF(rect);
            return NULL;
        }
    }

    return rect;
```